module3/ex8_math_practice/ex8_math_practice.py

```diff
@@ -1,9 +1,30 @@
-def create_math_problems():
-    pass
+import random
+MIN = -40
+MAX = 40
+NR_LINES = 100
+WIDTH = max(len(str(MIN)), len(str(MAX)))+2
 
 
-def solve_math_problems():
-    pass
+def create_math_problems(tmp_path):
+    my_file = open(tmp_path, 'w')
+    index = 0
+
+    for one_line in range(NR_LINES):
+        index += 1
+        nr_1 = random.randint(MIN, MAX)
+        nr_2 = random.randint(-40, 40)
+        nr_3 = random.randint(-40, 40)
+        nr_4 = random.randint(-40, 40)
+        rand_sign = "+" if random.randint(0, 1) == 0 else "-"
+        line = "["+str(index).rjust(3)+"]"+str(nr_1).rjust(5)+" "+rand_sign+" " + "("+str(nr_2).rjust(4)+")" + " " + rand_sign+" " + "("+str(nr_3).rjust(4)+")" + " "+rand_sign+" " + "("+str(nr_4).rjust(4)+") = ______\n"
```

Comment:
- For generating a random sign the best way I see would be using random.choice(['+', '-']).  I know I suggested to remove the lambda function for rand_sign, but missed that in that case the same generated operator will be the same on the entire generated line as generation is done only once on code line 18. Let's use random.choice and call it for each sign on the generated line.
- Line 19 is too long and does not respect PEP 8 maximum allowed length, write it in easier readable way.
- Whitespaces should be used around operators as per PEP8, exception is '=' operator on named parameters.
- I know I suggested using something like rjust and ljust, but I think better is to use f-string for numbers indentation and width, something like f"{number:{width}}", this seems easier to read.

Reply:
The WIDTH constant-like variable should have been used for formatting the numbers, i.e. the number of characters it fills.
line = f"[{index:{3}}] {random.choice(['+', '-'])} ({nr_1:{WIDTH}})"\
 f" {random.choice(['+', '-'])} ({nr_2:{WIDTH}})"\
 f" {random.choice(['+', '-'])} ({nr_3:{WIDTH}})"\
 f" {random.choice(['+', '-'])} ({nr_4:{WIDTH}}) = ______\n""